kolibri_explore_plugin/viewsets.py

```diff
@@ -0,0 +1,28 @@
+from kolibri.core.api import ReadOnlyValuesViewset
+from kolibri.core.content.api import ContentNodeViewset
+from kolibri.core.content.models import ContentNode
+from rest_framework.decorators import action
+from rest_framework.response import Response
+
+from .models import ContentNodeExtras
+from .models import ExternalContentTag
+
+
+class ExternalContentTagViewset(ReadOnlyValuesViewset):
+    values = ("id", "tag_name")
+
+    def get_queryset(self):
+        return ExternalContentTag.objects.all()
+
+
+class ContentNodeExtrasViewset(ContentNodeViewset):
+    @action(detail=False)
+    def by_external_tag(self, request):
+        tag = self.request.query_params.get("tag", None)
+        queryset = self.filter_queryset(self.get_queryset())
+        matching_extras = ContentNodeExtras.objects.filter(tags__tag_name=tag)
+        content_ids = [extra.content_node.pk for extra in matching_extras]
+        queryset = queryset & ContentNode.objects.filter_by_uuids(
+            content_ids, validate=False
+        )
```

Comment:
`ContentNode.objects.filter_by_uuids` accepts a QuerySet as a parameter, which would be more efficient than building a list here.

```suggestion
        queryset = queryset & ContentNode.objects.filter_by_uuids(
            matching_extras.values_list('content_node'), validate=False
        )
```

If we're feeling ambitious, we can even skip that `values_list` part because ContentNodeExtras is keyed by the content_node field:

```suggestion
        queryset = queryset & ContentNode.objects.filter_by_uuids(
            matching_extras, validate=False
        )
```

Reply:
Ah that's much better! Indeed keeping things in querysets is the proper way of querying the database. I copied the code from another plugin which probably needs this improvement too. I'll try to find it!